tasks/RTPTask.cpp

```diff
@@ -0,0 +1,365 @@
+/* Generated from orogen/lib/orogen/templates/tasks/Task.cpp */
+
+#include <base-logging/Logging.hpp>
+#include <gst/gstcaps.h>
+#include <set>
+
+#include "Helpers.hpp"
+#include "RTPTask.hpp"
+
+using namespace std;
+using namespace gstreamer;
+using namespace base::samples::frame;
+
+RTPTask::RTPTask(std::string const& name)
+    : RTPTaskBase(name)
+{
+}
+
+RTPTask::~RTPTask()
+{
+}
+
+/// The following lines are template definitions for the various state machine
+// hooks defined by Orocos::RTT. See Task.hpp for more detailed
+// documentation about them.
+
+RTPSessionStatistics RTPTask::extractRTPSessionStats(GstElement* session)
+{
+    RTPSessionStatistics session_stats;
+
+    GstStructure* gst_stats = nullptr;
+    g_object_get(session, "stats", &gst_stats, NULL);
+    LOG_INFO_S << "stats: " << gst_structure_to_string(gst_stats);
```

Comment:
This is debug stuff (i.e. OK for development, but should really not be in git). It would fill GB of logs in little time.